hermes/probe/create/create.go

```diff
@@ -168,25 +167,26 @@ func CreateFile(ctx context.Context, target *probe.Target, fileID int32, fileSiz
 	target.LatencyMetrics.APICallLatency[metrics.APICreateFile][status].Metric(hermesAPILatencySeconds).AddFloat64(time.Now().Sub(start).Seconds())
 
 	// Verify that the file that has just been created is in fact present in the target system
-	query := &storage.Query{Prefix: fmt.Sprintf(FileNameFormat, fileID, "")}
+	fileNamePrefix := fmt.Sprintf(FileNameFormat, fileID, "")
+	query := &storage.Query{Prefix: fileNamePrefix}
 	start = time.Now()
 	objIter := client.Bucket(bucketName).Objects(ctx, query)
-	finish := time.Now()
 	obj, err := objIter.Next()
-	if err != nil {
-		target.LatencyMetrics.APICallLatency[metrics.APIListFiles][metrics.FileMissing].Metric(hermesAPILatencySeconds).AddFloat64(finish.Sub(start).Seconds())
-		return fmt.Errorf("CreateFile check failed: %w", err)
-	}
-	namesFound := []string{obj.Name}
+	var namesFound []string
 	for {
 		if obj == nil {
 			break
 		}
 		namesFound = append(namesFound, obj.Name)
 		obj, err = objIter.Next()
 	}
+	finish := time.Now()
+	if len(namesFound) == 0 {
+		target.LatencyMetrics.APICallLatency[metrics.APIListFiles][metrics.FileMissing].Metric(hermesAPILatencySeconds).AddFloat64(finish.Sub(start).Seconds())
+		return fmt.Errorf("CreateFile check failed: %w", err)
```

Comment:
There's no underlying error here.
What fails here is that we expected exactly one result from the query but got none.
If there was an unrelated error (let's say a network issue), it should have been caught in the loop above.

Reply:
Maybe I'm misunderstanding sorry but isn't the difference the fact that if there are 0 files then for sure our create file operation must have failed and not create a file and if there is more than one file it could be another issue?

Reply:
There are 4 different outcomes
1) The query itself fails due to some error
2) We get zero results
3) We get one result
4) We get more than one result

#1 is not handled by the current code.
I would re-arrange the code to follow the pattern in n https://godoc.org/cloud.google.com/go/storage#hdr-Listing_objects.
Then a query failure will be handled.

Reply:
I will I just wonder whether we should distinguish between 2 and 4

Reply:
We must distinguish #2 and #4: if we get zero results here that's a signal that creating the file failed silently (we didn't get any error when we finished writing the file on line 161.)
If we get one result (and the filename matches our expectation), the creation succeeded.

Reply:
Distinguishing between #2 and #4 now